File: 백준/Silver/15650. N과 M （2）/N과 M （2）.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split(' ');
const N = Number(input[0]);
const M = Number(input[1]);
const result = [];

function dfs(start, length) {
    // 종료 : M개만큼 선택한 경우
    if(length === M) {
        console.log(result.join(' '));
        return;
    }
    
    // 현재값보다 큰 값들을 dfs 인자로 사용
    for(let i=start; i<=N; i++) {
        result.push(i);
        dfs(i+1, length+1);
        
        result.pop(); // 다시 result 배열 원소 제거
    }
}

dfs(1,0);

File: 백준/Silver/15650. N과 M （2）/N과 M （2）.py
import sys
input = sys.stdin.readline

N, M = map(int, input().strip().split())

def recursive_find_sequence(depth, temp_result, visited):
    # 기저조건
    if depth == M:
        print(' '.join(map(str, temp_result)))
        return

    for next_node in range(1, N + 1):
        if (not visited[next_node]) and (temp_result[-1] < next_node):
            visited[next_node] = True
            temp_result.append(next_node)
            recursive_find_sequence(depth + 1, temp_result, visited)
            visited[next_node] = False
            temp_result.pop()

# 실행
for node in range(1, N + 1):
    visited = [False] * (N + 1)
    visited[node] = True
    recursive_find_sequence(1, [node], visited)